modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/test/PublicationUserNotificationHandlerTest.java

```diff
@@ -0,0 +1,240 @@
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.test.rule.FeatureFlags;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+
+import java.util.List;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+
+/**
+ * @author Gislayne Vitorino
+ * @author Brooke Dalton
+ */
+@FeatureFlags("LPD-11018")
+@RunWith(Arquillian.class)
+public class PublicationUserNotificationHandlerTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@AfterClass
+	public static void tearDownClass() throws Exception {
+		Bundle bundle = FrameworkUtil.getBundle(JournalArticleService.class);
+
+		BundleContext bundleContext = bundle.getBundleContext();
+
+		for (Bundle curBundle : bundleContext.getBundles()) {
+			if (_BUNDLE_SYMBOLIC_NAME.equals(curBundle.getSymbolicName())) {
+				if (curBundle.getState() != Bundle.ACTIVE) {
+					curBundle.start();
+				}
+
+				break;
+			}
+		}
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+	}
+
+	@Test
+	public void testScheduledPublicationConflictFails() throws Exception {
+		JournalArticle journalArticle = JournalTestUtil.addArticle(
+			_group.getGroupId(), RandomTestUtil.randomString(),
+			StringPool.BLANK);
+
+		CTCollection ctCollection = _ctCollectionLocalService.addCTCollection(
+			null, TestPropsValues.getCompanyId(), TestPropsValues.getUserId(),
+			0, RandomTestUtil.randomString(), null);
+
+		try (SafeCloseable safeCloseable =
+				CTCollectionThreadLocal.setCTCollectionIdWithSafeCloseable(
+					ctCollection.getCtCollectionId())) {
+
+			journalArticle = _journalArticleLocalService.updateArticle(
+				journalArticle.getId(), RandomTestUtil.randomString());
+		}
+
+		try (SafeCloseable safeCloseable2 =
+				CTCollectionThreadLocal.setProductionModeWithSafeCloseable()) {
+
+			_journalArticleLocalService.deleteArticle(
+				_group.getGroupId(), journalArticle.getArticleId(),
+				ServiceContextTestUtil.getServiceContext());
+		}
+
+		_ctCollectionService.publishCTCollection(
+			TestPropsValues.getUserId(), ctCollection.getCtCollectionId());
+
+		List<UserNotificationEvent> userNotificationEvents =
+			_userNotificationEventLocalService.getUserNotificationEvents(
+				TestPropsValues.getUserId());
+
+		for (UserNotificationEvent userNotificationEvent :
+				userNotificationEvents) {
+
+			String notificationEventType = userNotificationEvent.getType();
```

Comment:
Skip if the notification type is not publications

```
if (Objects.equals(CTPortletKeys.PUBLICATIONS, userNotificationEvent.getType()) {
    continue;
}
```
I prefer using `Objects.equals` for strings comparison to avoid possible NPE.